### PR TITLE
feat(mms): Add enhancedFetch-using grade client

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

* Fetch compulsory/grade/primary school data using enhancedFetch
* Updated grahpql modelling
* Mountebank mocks

## Why

* Old outdated client
* No dev data
* Confusing graphql models

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced `EducationServiceV2` for managing educational data related to family careers.
  - Enhanced the `GradeClientService` for improved retrieval of student assessments and family member data.
  - Added new models and interfaces for handling educational data, including `StudentCareer`, `CompulsorySchoolGradeLevelExamResults`, and `GradeDetail`.

- **Documentation**
  - Added a README with instructions for using the new grade client library.

- **Configuration**
  - Updated various configuration files to support the new grade client library.
  - Introduced OpenAPI specifications for the Grade API, detailing endpoints and schemas.

- **Chores**
  - Added code ownership for the new grade client library.

- **Tests**
  - Included Jest configurations and TypeScript settings for the new grade client library.

### CODE REVIEW

**Review by robertaandersen — APPROVED** [review]
LGTM

File: libs/api/domains/education/src/lib/educationMapper.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+export const mapCareer = (
+  data: StudentAssessmentsDto,
+): StudentCareer | undefined => {
+  const { firstAssessmentYear, lastAssessmentYear } =
+    data.assessmentsOverview.assessmentsYearSpan
+
+  return {
+    nationalId: data?.nationalId,
+    name: data?.name,
+    isChildOfUser: false,
```
Comment (thordurhhh) [inline libs/api/domains/education/src/lib/educationMapper.ts:14]: 👀

File: libs/api/domains/education/src/lib/models/gradeCategoryWeighted.model.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+@ObjectType('EducationCompulsorySchoolGradeCategoryWeighted', {
+  implements: () => GradeCategory,
+})
+export class GradeCategoryWeighted implements GradeCategory {
+  @Field()
+  label!: string
+
+  @Field(() => Grade, { nullable: true })
+  grade!: Grade
```
[the remark this answers is not in the document]
Comment (thordurhhh) [reply libs/api/domains/education/src/lib/models/gradeCategoryWeighted.model.ts:13]: Yes coderabbit. Good point.

@thorkellmani 👀

**Review by thordurhhh — APPROVED** [review]
Nice. Just the tiny little thing with isChildOfUser in mapper.

**Review by birkirkristmunds — APPROVED** [review]
No changes in our code.
LGTM.

File: libs/api/domains/education/src/lib/models/competence.model.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { Field, ObjectType } from '@nestjs/graphql'
+
+@ObjectType('EducationCompulsorySchoolCourseCompetence')
+export class CourseCompetence {
+  @Field() //B+
+  competencyGrade!: string
+
+  @Field({ nullable: true }) // A1
+  competenceStatus?: string
```
Comment (kksteini) [inline libs/api/domains/education/src/lib/models/competence.model.ts:9]: Is it possible to expound on these comments (or remove them)?

File: libs/api/domains/education/src/lib/models/course.model.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { Field, ObjectType } from '@nestjs/graphql'
+import { CourseCompetence } from './competence.model'
+import { GradeCategory } from './gradeCategory.model'
+import { Grade } from './grade.model'
+
+@ObjectType('EducationCompulsorySchoolCourse')
+export class Course {
+  @Field() //t.d. "Enska"
```
Comment (kksteini) [inline libs/api/domains/education/src/lib/models/course.model.ts:8]: Sorry for the nitpick, but I think the coding standard is to keep things in English.
```suggestion
  @Field() //f.x. "Enska"
```

File: libs/api/domains/education/src/lib/models/gradeCategory.model.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  resolveType(res) {
+    if (res.text) {
+      return GradeCategoryText
+    }
+
+    return GradeCategoryWeighted
+  },
+})
+export abstract class GradeCategory {
+  @Field() //t.d. "Lesskilningur"
```
Comment (kksteini) [inline libs/api/domains/education/src/lib/models/gradeCategory.model.ts:15]: Same here

**Review by kksteini — APPROVED** [review]

File: libs/api/domains/education/src/lib/models/course.model.ts (hunk 4 above)
Comment (thorkellmani) [reply libs/api/domains/education/src/lib/models/course.model.ts:8]: good eye

File: libs/api/domains/education/src/lib/models/competence.model.ts (hunk 3 above)
Comment (thorkellmani) [reply libs/api/domains/education/src/lib/models/competence.model.ts:9]: Should be removed

File: libs/api/domains/education/src/lib/models/gradeCategory.model.ts (hunk 5 above)
Comment (thorkellmani) [reply libs/api/domains/education/src/lib/models/gradeCategory.model.ts]: removed

File: libs/api/domains/education/src/lib/models/course.model.ts (hunk 4 above)
Comment (thorkellmani) [reply libs/api/domains/education/src/lib/models/course.model.ts]: removed

**Review by eirikurn — APPROVED** [review]
Approving core files :+1

**Pull request merged**